rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -6551,6 +6553,318 @@ private void assertRefundInProcessPegInVersionLegacy(
+
+        BtcBlockStoreWithCache btcBlockStore = mock(BtcBlockStoreWithCache.class);
+        BtcBlockStoreWithCache.Factory mockFactory = mock(BtcBlockStoreWithCache.Factory.class);
+        when(mockFactory.newInstance(repository, bridgeConstants, provider, activations)).thenReturn(btcBlockStore);
+        Block executionBlock = Mockito.mock(Block.class);
+        when(executionBlock.getNumber()).thenReturn(10L);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+                .withProvider(provider)
+                .withBridgeConstants(bridgeConstants)
+                .withActivations(activations)
+                .withBtcBlockStoreFactory(mockFactory)
+                .withExecutionBlock(executionBlock)
+                .withRepository(repository)
+                .build();
+        bridgeSupport = spy(bridgeSupport);
+
+        Keccak256 derivationArgumentsHash = PegTestUtils.createHash3(0);
+        Keccak256 fastBridgeDerivationHash = bridgeSupport.getFastBridgeDerivationHash(
+                derivationArgumentsHash,
+                userRefundBtcAddress,
+                lpBtcAddress,
+                lbcAddress
+        );
+
+        if (includeActiveFederation) {
+            Address activeFederationAddress = getFastBridgeAddressFromRedeemScript(
+                    activeFederation.getRedeemScript(),
+                    Sha256Hash.wrap(fastBridgeDerivationHash.getBytes())
+            );
+            tx.addOutput(valueToSend, activeFederationAddress);
+        }
+
+        if (includeRetiringFederation) {
+            Address retiringFederationAddress = getFastBridgeAddressFromRedeemScript(
+                    retiringFederation.getRedeemScript(),
+                    Sha256Hash.wrap(fastBridgeDerivationHash.getBytes())
+            );
+            tx.addOutput(valueToSend, retiringFederationAddress);
+        }
+        tx.addInput(
+                Sha256Hash.wrap(fastBridgeDerivationHash.getBytes()),
+                0, ScriptBuilder.createInputScript(null, srcKey));
+
+        List<Sha256Hash> hashes = new ArrayList<>();
+        hashes.add(tx.getHash());
+        PartialMerkleTree pmt = new PartialMerkleTree(bridgeConstants.getBtcParams(), bits, hashes, 1);
+        Sha256Hash merkleRoot = pmt.getTxnHashAndMerkleRoot(new ArrayList<>());
+        co.rsk.bitcoinj.core.BtcBlock registerHeader = new co.rsk.bitcoinj.core.BtcBlock(
+                bridgeConstants.getBtcParams(),
+                1,
+                PegTestUtils.createHash(2),
+                merkleRoot,
+                1,
+                1,
+                1,
+                new ArrayList<BtcTransaction>()
+        );
+        // simulate blockchain
+        mockChainOfStoredBlocks(
+                btcBlockStore,
+                registerHeader,
+                height + bridgeConstants.getBtc2RskMinimumAcceptableConfirmations(),
+                height
+        );
+
+        InternalTransaction rskTx = new InternalTransaction(
```

Comment:
Maybe you could consider mocking this also, to simplify

Reply:
You mean do this 
```suggestion
        InternalTransaction rskTx = mock(InternalTransaction.class);
```
If I do this, all the test fail because this code below will return always `FAST_BRIDGE_UNPROCESSABLE_TX_NOT_CONTRACT_ERROR_CODE`.

![image](https://user-images.githubusercontent.com/94651730/154729958-79afe1e3-6888-4595-afd8-dcba17d106f8.png)

This is needed, it is part of the dependencies needed for these tests.

Reply:
You are right, I see no workaround here. So let's just keep it with the actual class. Thanks for the clarification